packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -420,12 +420,19 @@ export class ApolloServerBase {
         .catch(err => {
           // We intentionally do not re-throw the exact error from the gateway
           // configuration as it may contain implementation details and this
-          // error will propagate to the client. We will, however, log the error
-          // for observation in the logs.
-          const message = "This data graph is missing a valid configuration.";
```

Comment:
This should remain the as the externally facing error.

Reply:
Several users have gotten confused with this externally facing error to the point of spending several days trying to dig into it more. I don't think the message here makes a lot of sense personally. What's the rationale behind maintaining it as the externally facing error message?

Reply:
We intentionally do not blindly surface all error details to the client.  That would be leaky to any errors that include sensitive details including configuration details about graphs and variants which belongs in the server's logs.  The user that you are referring to was receiving an instruction to check the logs in the error message, specifically:

```
This data graph is missing a valid configuration. More details may be available in the server logs.
```

They then went on to reveal that they were seeing a confusing GCS error message that revealed `AccessDenied`.  We should trap and improve that specific error message under the specific status code 403, since it's a known condition.  We should maintain a clear distinction over what is external and what goes in the logs.

Reply:
That's to say, putting things in the logs would have worked for this user, if it wasn't XML.  Let's fix that XML error and make it useful (within the Gateway's `loadFromRemoteStorage`, but not risk revealing any and all errors to the client.

Reply:
Definitely agreed :) this was a rather heavy-handed approach. The question I was asking is if we're attached to the phrase "this data graph is missing a valid configuration" since I think we can provide better context at the top level than that.